feat: update grading workflows and refresh tool versions

FidelusAleksander opened the pull request:
### Summary
<!-- A clear and concise description of what the problem or opportunity is. -->

Changed the way we do grading workflows, now using https://github.com/skills/action-keyphrase-checker
Removed skills/response-templates references
Updated all `action-text-variables` refs to v2 and yaml style vars
Added a tip to approve workflow run triggered by copilot review


### Changes
<!-- Describe the changes this pull request introduces. -->


<!-- If there's an existing issue for your change, please link to it below next to "Closes".
If there's _not_ an existing issue, please open one first to make it more likely that this update will be accepted. -->

Closes:

### Task list

- [ ] For workflow changes, I have verified the Actions workflows function as expected.
- [ ] For content changes, I have reviewed the [style guide](https://github.com/github/docs/blob/main/contributing/content-style-guide.md).

chriswblake commented on .github/workflows/4b-copilot-on-github.yml:
```diff
@@ -2,11 +2,11 @@ name: Step 4b # Copilot on GitHub
-  #     - main
-  #   types:
-  #     - edited
+  pull_request:
```
I originally disabled this workflow because we learned during beta testing that users with Copilot Free were getting blocked here. They couldn't used Copilot free. The simple answer was to disable this trigger. Is that different now?

aleqq123 replied:
I just went through the exercise on a free account, good catch

The PR summaries and copilot review are both not available on free accounts. So I imagine the feedback was something along the lines "I edited the PR myself, got 2 false checks that I can't get through"

I will disable this trigger

FidelusAleksander replied:
I just went through the exercise on a free account, good catch

The PR summaries and copilot review are both not available on free accounts. So I imagine the feedback was something along the lines "I edited the PR myself, got 2 false checks that I can't get through"

I will disable this trigger

chriswblake reviewed (approved):
Everything looks good. Just confirm if the trigger for workflow 4b should stay disabled or it is safe to enable now.

arilivigni reviewed (approved):
LGTM

Are we sticking with `--edit-last` or adding a new comment.
```
          gh issue comment "$ISSUE_URL" \
            --body-file exercise-toolkit/markdown-templates/step-feedback/checking-work.md \
            --edit-last
```

FidelusAleksander commented:
> LGTM
> 
> Are we sticking with `--edit-last` or adding a new comment.
> 
> ```
>           gh issue comment "$ISSUE_URL" \
>             --body-file exercise-toolkit/markdown-templates/step-feedback/checking-work.md \
>             --edit-last
> ```

I think a combination of both is probably good. Some comments make sense to be updated, some not e.g 
- we don't want the result table commented separately every time the grading job fails for a given step, so we use --edit-last
  ```yaml
      - name: Create comment - step results
        run: |
          gh issue comment "$ISSUE_URL" \
            --body "$COMMENT_BODY" \
            --edit-last
        env:
          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
          COMMENT_BODY: ${{ steps.build-message-step-results.outputs.updated-text }}
  ```
- It's probably best that if the grading workflow succeeds we don't override the results table with --edit-last so the exercise taker can see these results, in that case we would create a new comment
  ```yaml
        - name: Update comment - step finished
          run: |
            gh issue comment "$ISSUE_URL" \
              --body "$ISSUE_BODY"
          env:
            GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
            ISSUE_BODY: ${{ steps.build-message-step-finish.outputs.updated-text }}
  ```
  
All open to change, and probably standardize/document
